internal/cluster/validations/validations.go

```diff
@@ -420,6 +420,33 @@ func ValidateIPAddresses(ipV6Supported bool, obj interface{}) error {
 	if err != nil {
 		return err
 	}
+
+	switch c := obj.(type) {
+	case *models.ClusterCreateParams:
+		targetConfiguration := common.Cluster{}
+		targetConfiguration.ClusterNetworks = c.ClusterNetworks
+		targetConfiguration.ServiceNetworks = c.ServiceNetworks
+		targetConfiguration.MachineNetworks = c.MachineNetworks
+
+		api := ""
+		if c.APIVip != nil {
+			api = *c.APIVip
+		}
+
+		dualStack := network.CheckIfClusterIsDualStack(&targetConfiguration)
+		if dualStack {
+			err = network.VerifyVips(nil, string(c.MachineNetworks[0].Cidr), api, c.IngressVip, true, nil)
```

Comment:
Are machine networks required during create ?

Reply:
No, that's a good catch, thanks! We allow creating a cluster object without any Machine Networks, so this check needs to be hardened against it.